File: exercism/javascript/rna-transcription/rna-transcription.js
// solved on may 10, 2017

DnaTranscriber = function(){}

var builder = function(dna){
  strand = '';
  for (var i = 0 ; i <= dna.length - 1; i++)
  {
    letter = dna[i];
    if (letter !== 'C' && letter !== 'G' && letter !== 'A' && letter !== 'T') throw new Error('Invalid input');
    else {
      strand += (letter === 'C') ? 'G' :
        (letter === 'G') ? 'C' :
        (letter === 'A') ? 'U' :
        (letter === 'T') ? 'A' : '';
      }
    }
  return strand;
}

DnaTranscriber.prototype.toRna = function(dna) {
  return builder(dna, 0);
}

module.exports = DnaTranscriber;

File: exercism/python/rna-transcription/rna_transcription.py
#Solved aug 8, 2017

def to_rna(dna):
    rna  = ''    
    pairs = {'G':'C', 
             'C':'G', 
             'T':'A', 
             'A':'U' }
    def getComp(base):
        return pairs.get(base, '')

    for x in dna:
        if (x not in pairs.keys()):
            return ''
        else:
            rna += getComp(x)
    return rna
            
